pkg/controller/servicebindingrequest/annotations/attribute.go

```diff
@@ -0,0 +1,57 @@
+package annotations
+
+import (
+	"github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/nested"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const AttributeValue = "binding:env:attribute"
+
+// AttributeHandler handles "binding:env:attribute" annotations.
+type AttributeHandler struct {
+	// inputPath is the path that should be extracted from the resource.
+	inputPath string
+	// outputPath is the path the extracted data should be placed under in the
+	// resulting unstructured object in Handler.
+	outputPath string
+	// resource is the unstructured object to extract data using inputPath.
+	resource unstructured.Unstructured
+}
+
+// OutputPath returns the path the extracted data should be placed under.
+func (a *AttributeHandler) OutputPath() string {
+	if len(a.outputPath) > 0 {
+		return a.outputPath
+	}
+	return a.inputPath
+}
+
+// Handle returns a unstructured object according to the "binding:env:attribute"
+// annotation strategy.
+func (h *AttributeHandler) Handle() (Result, error) {
+	val, _, err := nested.GetValue(h.resource.Object, h.inputPath, h.OutputPath())
+	if err != nil {
+		return Result{}, err
```

Comment:
That construction can only be used when returning by reference, not by value like it currently is.

Reply:
Thanks, any reason why we cannot change the function signature to 

```
func (h *AttributeHandler) Handle() (*Result, error) {
```

That would avoid creating empty unnecessary structs (and lower the memory pressure) in case of errors.

Reply:
There are no tests to test the error scenario

Reply:
do we have an issue created for this @DhritiShikhar ?

Reply:
Increase coverage in existing tests #471

Reply:
> Thanks, any reason why we cannot change the function signature to
> 
> ```
> func (h *AttributeHandler) Handle() (*Result, error) {
> ```
> 
> That would avoid creating empty unnecessary structs (and lower the memory pressure) in case of errors.

Please send a PR to address this.